scrape/target.go

```diff
@@ -110,13 +110,17 @@ func (t *Target) Labels() labels.Labels {
 
 // DiscoveredLabels returns a copy of the target's labels before any processing.
 func (t *Target) DiscoveredLabels() labels.Labels {
+	t.mtx.Lock()
```

Comment:
Does this need to be a write lock?

Reply:
A better questions is why this is a RWMutex, which I don't see a need for here.

Reply:
good point I didn't even see it was a RWMutex. It looks like a leftover from some other code..